# Initial Cleaning

In [15]:
# load the dataset and insepct structure
import pandas as pd

teaser = pd.read_csv('teaser_comments.csv')

# drop rows missing comment text and/or author
teaser = teaser.dropna(subset=["text", "author"])

# fill in missing likeCOunt with 0
teaser['likeCount'] = teaser['likeCount'].fillna(0)

# print(teaser.head())
print(teaser.info())
print(teaser.describe(include = 'all'))

<class 'pandas.core.frame.DataFrame'>
Index: 33086 entries, 0 to 33091
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   group        33086 non-null  object 
 1   video_id     33086 non-null  object 
 2   publishedAt  33086 non-null  object 
 3   author       33086 non-null  object 
 4   text         33086 non-null  object 
 5   likeCount    33086 non-null  float64
dtypes: float64(1), object(5)
memory usage: 1.8+ MB
None
         group     video_id           publishedAt     author   text  \
count    33086        33086                 33086      33086  33086   
unique      10           10                 20868      24011  29049   
top     Kep1er  o5JSBiEfQR0  2021-12-28T15:01:06Z  @ning0105    OMG   
freq     11487        11487                    35         84    208   
mean       NaN          NaN                   NaN        NaN    NaN   
std        NaN          NaN                   NaN        NaN    NaN   
min    

After dropping relevant NaNs, our dataset has a total of 33,086 comments from teaser videos. I notice that there are 29,049 unique comments by 24,011 unqiue authors. This tells us that there are some authors that may post the same comment multiple times. To account for this, we will remove dpulicate comments that are written by the same author for the same video.

In [ ]:
before = len(teaser)

# drop duplicates for rows that have the same video id, author, and text
teaser = teaser.drop_duplicates(subset=["video_id", "author", "text"], keep="first")

# find how many duplicates were removed
after = len(teaser)
print(f"Removed {before - after} duplicate comments")


Removed 381 duplicate comments


Now, we've removed 381 duplicates for comments that have the same content, written by the same author, and for the same video. That should put us a total of 32,705 teaser comments.

In [17]:
len(teaser)

32705

In [25]:
# read the cleaned data into a new csv
teaser.to_csv('clean_teaser_comments.csv', index = False)

# Basic Statistical Analysis

In [19]:
# comment counts per group
group_counts = teaser["group"].value_counts()
print(group_counts)

group
Kep1er         11318
ZEROBASEONE     7338
NMIXX           5600
LE SSERAFIM     2863
IVE             2723
TNX             1178
RIIZE           1050
BOYNEXTDOOR      362
H1-KEY           152
YOUNITE          121
Name: count, dtype: int64


I notice that groups that are more popular generally have higher comment counts

In [23]:
# most liked comments
top_comments = teaser.sort_values("likeCount", ascending=False).head(10)
print(top_comments[["group", "likeCount", "text"]])

             group  likeCount  \
16363       Kep1er    26548.0   
7592         NMIXX     9742.0   
20976       Kep1er     7202.0   
6397         NMIXX     6935.0   
2489   LE SSERAFIM     6652.0   
5319           IVE     5753.0   
9353         NMIXX     5508.0   
6244         NMIXX     5066.0   
5631           IVE     4938.0   
1405   LE SSERAFIM     3890.0   

                                                    text  
16363                         They are finally debuting😭  
7592              OH MY FREAKY FISHY! THEY LOOK SO GOOD!  
20976  ALL THE BEST KEP1ER QUEENS ROCK IT MY FULL SUP...  
6397   Wait, is this an entirely different song? Are ...  
2489   It&#39;s like saying &quot;why is it too short...  
5319   Their debut is getting nearer and nearer, It&#...  
9353            It will be iconic, I CAN&#39;T WAITTT 🔥🔥  
6244   QUE TEASER FOI ESSE ??? JYP Finalmente sabendo...  
5631                        YOU MAKE ME FEEL LIKE ELEVEN  
1405   Obsessed with the fact that they are da

I notice that a majority of the comments with the most likes are positive, suggesting that it is likelt that the majority of teaser comments show positive sentiment. 

In [ ]:
# most common words
from collections import Counter
words = " ".join(teaser["text"]).lower().split()
Counter(words).most_common(20)

[('the', 3744),
 ('i', 3554),
 ('so', 3539),
 ('is', 2606),
 ('this', 2442),
 ('a', 1999),
 ('omg', 1916),
 ('for', 1790),
 ('to', 1767),
 ('wait', 1587),
 ('can&#39;t', 1510),
 ('debut', 1352),
 ('and', 1352),
 ('be', 1280),
 ('of', 1217),
 ('my', 1212),
 ('it', 1145),
 ('love', 1133),
 ('excited', 1034),
 ('da', 997)]

I notice while there are many stopwords here as the most common words, there are some words that show positive sentiment such as 'omg', 'love', and 'excited', suggesting that positive sentiment may be more prevalent in the teaser videos than negative sentiment. 